source/module_hamilt_pw/hamilt_ofdft/ml_tools/data.cpp

```diff
@@ -0,0 +1,321 @@
+#include "./data.h"
+#include "npy.hpp"
+
+Data::~Data()
+{
+    delete[] load_gammanl;
```

Comment:
It seems dangerous to delete here without knowing where is the 'new'

Reply:
They are initialized in the function Data::init_lable``.
![image](https://github.com/user-attachments/assets/35a5490c-4c09-4882-8432-71c90a5716d8)